### PR TITLE
Feature/database/332 database postgresql instead of sqlite3

### LINKED ISSUE #332: Database: Support PostgreSQL
PostgreSQL support should be added, and the choice of DBMS should be configurable

### PR DESCRIPTION
**Pull request by lukas0820**
closes #332

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -3,6 +3,25 @@ version: '3.8'
 name: manvsim
 
 services:
+
+  postgres:
+    image: postgres:latest
+    ports:
+      - 5432:5432
```
Comment (n00on) [inline docker-compose.yaml:10]: Its recommended to use quotes here. Also i think i would like it better if the postgres service was below app, server and web

File: docker-compose.yaml [hunk 2]
```diff
@@ -3,6 +3,25 @@ version: '3.8'
...
+    environment:
+      POSTGRES_USER: admin
+      POSTGRES_PASSWORD: t3zgVzcWTd3WV8KTmnNH
```
Comment (n00on) [inline docker-compose.yaml:13]: Is the database accessible from outside the machine/network?

**Review by n00on — CHANGES_REQUESTED** [review]

File: server/Dockerfile [hunk 3]
```diff
@@ -19,18 +18,12 @@ RUN find . -type f -print0 | xargs -0 dos2unix
...
-FROM setup AS debug
-
-ENV LOAD_TEST_DATA=true
+RUN pipenv install --deploy --ignore-pipfile || exit 1
+RUN pipenv install gunicorn
 
-EXPOSE 5002
-CMD pipenv run flask --app main --debug run --host=0.0.0.0 --port=5002 --cert=fullchain.pem --key=privkey.pem
 
 FROM setup AS prod
```
Comment (n00on) [inline server/Dockerfile:26]: If we decide to remove debug mode, we should adjust the README

File: server/scripts/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+pipenv run flask --app main db init
+pipenv run flask --app main db upgrade
+pipenv run python dbsetup.py
+pipenv run gunicorn -w 1 --threads 32 -b 0.0.0.0:5002 --certfile=fullchain.pem --keyfile=privkey.pem 'main:app'
```
[the remark this answers is not in the document]
Comment (n00on) [reply server/scripts/entrypoint.sh:6]: Not too much i guess, but document this setting :)

File: server/Dockerfile [hunk 5]
```diff
@@ -4,7 +4,6 @@ WORKDIR /usr/src/app
 # Install dependencies
 RUN apt-get update && \
     apt-get install -y \
-    make \
     dos2unix \
     libmagic1
```
Comment (n00on) [inline server/Dockerfile:9]: if we don't build the react app anymore we can also skip `COPY ../web web`

File: server/dbsetup.py [hunk 6]
```diff
@@ -255,16 +251,36 @@ def insert(data):
...
     __create_roles()
+    db.session.commit()
+
     __create_patients()
+    db.session.commit()
+
     __create_locations()
+    db.session.commit()
+
+    __create_players(db=db)
```
[the remark this answers is not in the document]
Comment (lukas0820) [reply server/dbsetup.py]: fixed

File: docker-compose.yaml (hunk 1 above)
Comment (lukas0820) [reply docker-compose.yaml]: done

File: server/Dockerfile (hunk 5 above)
Comment (lukas0820) [reply server/Dockerfile:9]: removed

File: server/scripts/entrypoint.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (lukas0820) [reply server/scripts/entrypoint.sh]: done

File: docker-compose.yaml (hunk 2 above)
Comment (lukas0820) [reply docker-compose.yaml]: moved to config file

File: server/Dockerfile (hunk 3 above)
Comment (lukas0820) [reply server/Dockerfile:25]: removed the debug mode for the server in readme

**Comment by lukas0820** [comment]
@YIllmann could you review the server changes of this PR?

**Pull request merged**